lib/matplotlib/backends/backend_pgf.py

```diff
@@ -36,55 +36,61 @@
 # which is not recognized by TeX.
 
 
-def get_fontspec():
-    """Build fontspec preamble from rc."""
-    latex_fontspec = []
-    texcommand = mpl.rcParams["pgf.texsystem"]
-
-    if texcommand != "pdflatex":
-        latex_fontspec.append("\\usepackage{fontspec}")
+@_api.caching_module_getattr
+class __getattr__:
+    NO_ESCAPE = _api.deprecated("3.6", obj_type="")(
+        property(lambda self: _NO_ESCAPE))
+    re_mathsep = _api.deprecated("3.6", obj_type="")(
+        property(lambda self: _split_math.__self__))
 
-    if texcommand != "pdflatex" and mpl.rcParams["pgf.rcfonts"]:
-        families = ["serif", "sans\\-serif", "monospace"]
-        commands = ["setmainfont", "setsansfont", "setmonofont"]
-        for family, command in zip(families, commands):
-            # 1) Forward slashes also work on Windows, so don't mess with
-            # backslashes.  2) The dirname needs to include a separator.
-            path = pathlib.Path(fm.findfont(family))
-            latex_fontspec.append(r"\%s{%s}[Path=\detokenize{%s}]" % (
-                command, path.name, path.parent.as_posix() + "/"))
 
-    return "\n".join(latex_fontspec)
+@_api.deprecated("3.6")
+def get_fontspec():
+    """Build fontspec preamble from rc."""
+    with mpl.rc_context({"pgf.preamble": ""}):
+        return _get_preamble()
 
 
+@_api.deprecated("3.6")
 def get_preamble():
     """Get LaTeX preamble from rc."""
     return mpl.rcParams["pgf.preamble"]
 
-###############################################################################
 
-# This almost made me cry!!!
-# In the end, it's better to use only one unit for all coordinates, since the
+def _get_preamble():
+    """Prepare a LaTeX preamble based on the rcParams configuration."""
+    preamble = [mpl.rcParams["pgf.preamble"]]
+    if mpl.rcParams["pgf.texsystem"] != "pdflatex":
+        preamble.append("\\usepackage{fontspec}")
+        if mpl.rcParams["pgf.rcfonts"]:
+            families = ["serif", "sans\\-serif", "monospace"]
+            commands = ["setmainfont", "setsansfont", "setmonofont"]
+            for family, command in zip(families, commands):
+                # 1) Forward slashes also work on Windows, so don't mess with
+                # backslashes.  2) The dirname needs to include a separator.
+                path = pathlib.Path(fm.findfont(family))
+                preamble.append(r"\%s{%s}[Path=\detokenize{%s}]" % (
+                    command, path.name, path.parent.as_posix() + "/"))
```

Comment:
We can hard-code the `/`:
```suggestion
                preamble.append(r"\%s{%s}[Path=\detokenize{%s/}]" % (
                    command, path.name, path.parent.as_posix()))
```

Reply:
done